test/general/implementation_details/test_sycl_iterator_operations.pass.cpp

```diff
@@ -0,0 +1,287 @@
+    EXPECT_FALSE(__internal::__iterators_possibly_equal(oneapi::dpl::begin(buf1), it2), "wrong __iterators_possibly_equal result");
+    EXPECT_FALSE(__internal::__iterators_possibly_equal(oneapi::dpl::begin(buf1), it2Ref),
+                 "wrong __iterators_possibly_equal result");
+    EXPECT_FALSE(__internal::__iterators_possibly_equal(oneapi::dpl::begin(buf1), oneapi::dpl::begin(buf2)),
+                 "wrong __iterators_possibly_equal result");
+    EXPECT_FALSE(__internal::__iterators_possibly_equal(it1, oneapi::dpl::begin(buf2)), "wrong __iterators_possibly_equal result");
+
+    EXPECT_FALSE(__internal::__iterators_possibly_equal(oneapi::dpl::begin(buf1), nullptr),
+                 "wrong __iterators_possibly_equal result");
+    EXPECT_FALSE(__internal::__iterators_possibly_equal(nullptr, oneapi::dpl::begin(buf2)),
+                 "wrong __iterators_possibly_equal result");
+
+    // sub - buffer vs it's "root" buffer (expect true)
+    sycl::buffer<int, 1> buf11{buf1, sycl::range<1>{0}, sycl::range<1>{0}};
+    EXPECT_TRUE(__internal::__iterators_possibly_equal(oneapi::dpl::end(buf1), oneapi::dpl::begin(buf11)),
+                "wrong __iterators_possibly_equal result");
+
+    // sub - buffer vs sub - buffer which share a "root" buffer(expect true)
+    sycl::buffer<int, 1> buf12{buf1, sycl::range<1>{0}, sycl::range<1>{0}};
+    EXPECT_TRUE(__internal::__iterators_possibly_equal(oneapi::dpl::begin(buf11), oneapi::dpl::end(buf12)),
+                "wrong __iterators_possibly_equal result");
+
+    // two sycl_iterators pointing to different elements in the same "root" buffer(expect false)
+    auto it1next = it1 + 1;
+    EXPECT_FALSE(__internal::__iterators_possibly_equal(it1, it1next), "wrong __iterators_possibly_equal result");
+
+    {
+        float floatData = .0;
+
+        std::vector<int> dataVec{1, 2, 3};
+        std::vector<int>::const_iterator intConstData = dataVec.cbegin();
+        std::vector<int>::iterator intData = dataVec.begin();
+
+        // check pointer + pointer
+        EXPECT_TRUE(__internal::__iterators_possibly_equal(intData, intData), "wrong __iterators_possibly_equal result");
+        // check const pointer + pointer
+        EXPECT_TRUE(__internal::__iterators_possibly_equal(intConstData, intData), "wrong __iterators_possibly_equal result");
+        // check pointer + const pointer
+        EXPECT_TRUE(__internal::__iterators_possibly_equal(intData, intConstData), "wrong __iterators_possibly_equal result");
+        // check pointer + pointer to other type
+        EXPECT_FALSE(__internal::__iterators_possibly_equal(intData, &floatData), "wrong __iterators_possibly_equal result");
+    }
+
+    {
+        std::vector<int> dataVec1{1, 2, 3};
+        std::vector<int> dataVec2{4, 5, 6};
+        EXPECT_FALSE(__internal::__iterators_possibly_equal(dataVec1.begin(), dataVec2.begin()), "wrong __iterators_possibly_equal result");
+    }
```

Comment:
If I understand our offline discussion correctly, this will throw with MS STL and debug due to comparing iterators from different containers.
Do we want to protect this with a "broken macro" for this case (especially since this is a unit test).  At this point, we don't really have a strategy for resolving this in that situation (other than removing the need to use this function from exclusive_scan). Do we want to pollute our test results with this known failure case?